pkg/resource/deploy/source_eval.go

```diff
@@ -2165,6 +2259,73 @@ type registerResourceEvent struct {
 	done chan *RegisterResult // the channel to communicate with after the resource state is available.
 }
 
+func (rm *resmon) CreateNewContext(
+	ctx context.Context,
+	req *pulumirpc.CreateNewContextRequest,
+) (*pulumirpc.CreateNewContextResponse, error) {
+	d := &defaultProviders{
+		defaultProviderInfo: rm.defaultProviders.defaultProviderInfo,
+		providers:           deepcopy.Copy(rm.defaultProviders.providers).(map[string]providers.Reference),
+		config:              rm.defaultProviders.config,
+		requests:            make(chan defaultProviderRequest),
+		providerRegChan:     rm.defaultProviders.providerRegChan,
+		cancel:              rm.defaultProviders.cancel,
+	}
+
+	newRm := &defaultProviderResmon{
```

Comment:
probably call this a subcontextResmon or similar, it's unlikely to stay just for default providers.

Reply:
Yeah, I'll rename it :+1: